src/core/ext/transport/chaotic_good/client_transport.cc

```diff
@@ -169,23 +183,30 @@ auto ChaoticGoodClientTransport::OnTransportActivityDone() {
 }
 
 ChaoticGoodClientTransport::ChaoticGoodClientTransport(
-    std::unique_ptr<PromiseEndpoint> control_endpoint,
-    std::unique_ptr<PromiseEndpoint> data_endpoint,
-    std::shared_ptr<grpc_event_engine::experimental::EventEngine> event_engine)
-    : outgoing_frames_(4),
-      transport_(std::move(control_endpoint), std::move(data_endpoint)),
-      writer_{
-          MakeActivity(
-              // Continuously write next outgoing frames to promise endpoints.
-              TransportWriteLoop(), EventEngineWakeupScheduler(event_engine),
-              OnTransportActivityDone()),
-      },
-      reader_{MakeActivity(
-          // Continuously read next incoming frames from promise endpoints.
-          TransportReadLoop(), EventEngineWakeupScheduler(event_engine),
-          OnTransportActivityDone())} {}
+    PromiseEndpoint control_endpoint, PromiseEndpoint data_endpoint,
+    const ChannelArgs& args,
+    std::shared_ptr<grpc_event_engine::experimental::EventEngine> event_engine,
+    HPackParser hpack_parser, HPackCompressor hpack_encoder)
+    : allocator_(args.GetObject<ResourceQuota>()
+                     ->memory_quota()
+                     ->CreateMemoryAllocator("chaotic-good")),
+      outgoing_frames_(4) {
+  auto transport = MakeRefCounted<ChaoticGoodTransport>(
+      std::move(control_endpoint), std::move(data_endpoint),
+      std::move(hpack_parser), std::move(hpack_encoder));
+  writer_ = MakeActivity(
+      // Continuously write next outgoing frames to promise endpoints.
+      TransportWriteLoop(transport), EventEngineWakeupScheduler(event_engine),
+      OnTransportActivityDone());
+  reader_ = MakeActivity(
+      // Continuously read next incoming frames from promise endpoints.
+      TransportReadLoop(std::move(transport)),
+      EventEngineWakeupScheduler(event_engine), OnTransportActivityDone());
+  gpr_log(GPR_INFO, "ChaoticGoodClientTransport() %p", this);
```

Comment:
Probably this log statement should be removed.

Reply:
done